### PR TITLE
Moving S5 models to a folder and other bug fixes

### PR DESCRIPTION
**Pull request by taylorbell57**
This PR primarily resolves Issue #90 by moving the eureka/S5_lightcurve_fitting/models.py file to many files in a new eureka/S5_lightcurve_fitting/models folder. All of the old utility is maintained with an __init__.py file inside the new models folder, while the code is now easier to parse through and easier for others to contribute to.

In preparing this PR, I encountered Issue #94 where reload statements break my ability separate models.py into a models folder. For now, I have simply removed the reload statements, but this will need to be discussed in more detail in that issue.

I have also started using a new computer (MacOS 11.6 instead of Ubuntu 20.04) and had significant challenges using Eureka! until I made the included changes to the requirements.txt file. If someone else can make a new conda environment and make sure that this requirements list is correct and ideally platform independent that would be great. I also alphabetized the requirements.txt file to make it easier to read through (before it was hard to see if a requirement was present or not).

I also updated the main 51## figure to include a third panel showing the uncalibrated flux and detector*systematics model which can be very helpful in troubleshooting.

Finally, I made some other small tweaks like ensuring that the logs along the S2->S5 path all get appended to the logs from the previous stage. Previously the S3 log would not include the log from S2, but now it will do so as long as the Stage 2 outputs were made by Eureka and not `jwst`.

### CODE REVIEW

**Comment by taylorbell57** [comment]
Oh, and this PR also closes #92 for reasons I don't fully understand

File: eureka/S5_lightcurve_fitting/fitters.py [hunk 1]
```diff
@@ -11,10 +11,6 @@
 from .parameters import Parameters
 from .likelihood import computeRedChiSq, lnprob, ptform
 from . import plots_s5 as plots
-#FINDME: Keep reload statements for easy testing
```
Comment (kevin218) [inline eureka/S5_lightcurve_fitting/fitters.py:14]: Keep reload statements for easy testing

File: eureka/S5_lightcurve_fitting/lightcurve.py [hunk 2]
```diff
@@ -11,10 +11,6 @@
 from . import models as m
 from . import fitters as f
 from .utils import COLORS
-#FINDME: Keep reload statements for easy testing
```
Comment (kevin218) [inline eureka/S5_lightcurve_fitting/lightcurve.py:14]: Keep reload statements for easy testing

File: eureka/S5_lightcurve_fitting/s5_fit.py [hunk 3]
```diff
@@ -9,11 +9,6 @@
 from . import lightcurve as lc
 from . import models as m
 from .utils import get_target_data
-#FINDME: Keep reload statements for easy testing
```
Comment (kevin218) [inline eureka/S5_lightcurve_fitting/s5_fit.py:12]: Keep reload statements for easy testing

**Review by kevin218 — DISMISSED** [review]
Keep reload statements for easy testing

File: eureka/S5_lightcurve_fitting/fitters.py (hunk 1 above)
Comment (taylorbell57) [reply eureka/S5_lightcurve_fitting/fitters.py:14]: Did you see my issue at Issue #94 related to these reload statements?

**Comment by taylorbell57** [comment]
Alright, made those edits @kevin218 and it's ready to merge with your final approval

**Review by kevin218 — APPROVED** [review]

**Pull request merged**